ui/packages/catalog-ui-search/src/main/webapp/component/query-advanced/query-custom/query-custom.view.js

```diff
@@ -48,17 +54,39 @@ define([
                 this.queryAdvanced.currentView.deserialize(cql.simplify(cql.read(this.model.get('cql'))));
             }
 
+            if (ResultForm.getResultTemplatesProperties()) {
+                var detailLevelProperty = new Property({
+                    label: 'Detail Level',
+                    enum: ResultForm.getResultTemplatesProperties(),
+                    id: 'Detail Level'
+                });
+
+                this.listenTo(detailLevelProperty, 'change:value', this.handleChangeDetailLevel);
+                this.resultTemplate.show(new PropertyView({
+                    model: detailLevelProperty
+                }));
+
+                this.resultTemplate.currentView.turnOnLimitedWidth();
+                this.resultTemplate.currentView.turnOnEditing();
+            }
             this.querySettings.currentView.turnOffEditing();
             this.queryAdvanced.currentView.turnOffEditing();
             this.edit();
         },
+        handleChangeDetailLevel: function(model, values) {
+            $.each(model.get('enum') , (function(index, value) {
+                if (values[0] === value.value) {
+                    this.model.set('selectedResultTemplate', value);
```

Comment:
❗️ We're setting values on the query model here that aren't actually reflected in [Query.js](https://github.com/codice/ddf/blob/master/ui/packages/catalog-ui-search/src/main/webapp/js/model/Query.js#L74-L88)

Reply:
@mackncheesiest @andrewkfiedler Does it actually need to be reflected in Query.js?  It would be undefined in most cases

Reply:
That's the model that is used throughout the UI for queries. If we're setting or retrieving anything on that model, I would certainly like it to be _at least_ assigned a default value of undefined in Query.js even if only for documentation purposes for developers.

Reply:
We can add the default value of undefined for documentation purposes.